timemachine/fe/mcgregor.py

```diff
@@ -286,6 +292,31 @@ def atom_map_pop(map_1_to_2, map_2_to_1, idx, jdx):
     map_2_to_1[jdx] = UNMAPPED
 
 
+def _graph_is_disconnected(g, mapped_nodes, demapped_nodes, unvisited_nodes):
+    mapped_nodes = set(mapped_nodes)
+    demapped_nodes = set(demapped_nodes)
+    unvisited_nodes = set(unvisited_nodes)
+
+    # check intersections
+    assert mapped_nodes.union(demapped_nodes).union(unvisited_nodes) == set(range(g.n_vertices))
+    assert len(mapped_nodes.intersection(demapped_nodes)) == 0
+    assert len(mapped_nodes.intersection(unvisited_nodes)) == 0
+    assert len(demapped_nodes.intersection(unvisited_nodes)) == 0
+
+    all_possible_nodes = mapped_nodes.union(unvisited_nodes)
+
+    if len(mapped_nodes) > 0:
+        sg_all_possible = g.nxg.subgraph(all_possible_nodes)
+        sg_ccs = nx.connected_components(sg_all_possible)
+        # see if all the mapped nodes belong to the same connected component
+        for cc in sg_ccs:
+            if set(cc).intersection(mapped_nodes) == set(mapped_nodes):
```

Comment:
nit:
```suggestion
            if set(mapped_nodes).issubset(cc):
```
(or `set(mapped_nodes) <= set(cc)`)